src/BloomExe/Publish/Android/PublishToAndroidApi.cs

```diff
@@ -138,5 +141,61 @@ public static void ReportAnalytics(string mode, Book.Book book)
 		{
 			Analytics.Track("Publish Android", new Dictionary<string, string>() {{"mode", mode}, {"BookId", book.ID}, { "Country", book.CollectionSettings.Country}, {"Language", book.CollectionSettings.Language1Iso639Code}});
 		}
+
+		/// <summary>
+		/// This is the core of sending a book to a device. We need a book and a bookServer in order to come up
+		/// with the .bloomd file.
+		/// We are either simply saving the .bloomd to destFileName, or else we will make a temporary .bloomd file and
+		/// actually send it using sendAction.
+		/// We report important progress on the progress control. This includes reporting that we are starting
+		/// the actual transmission using startingMessageAction, which is passed the safe file name (for checking pre-existence
+		/// in UsbPublisher) and the book title (typically inserted into the message).
+		/// If a confirmAction is passed (currently only by UsbPublisher), we use it check for a successful transfer
+		/// before reporting completion (except for file save, where the current message is inappropriate).
+		/// This is an awkward case where the three ways of publishing are similar enough that
+		/// it's annoying and dangerous to have three entirely separate methods but somewhat awkward to combine them.
+		/// Possibly we could eventually make them more similar, e.g., it would simplify things if they all said
+		/// "Sending X to Y", though I'm not sure that would be good i18n if Y is sometimes a device name
+		/// and sometimes a path.
+		/// </summary>
+		/// <param name="book"></param>
+		/// <param name="destFileName"></param>
+		/// <param name="sendAction"></param>
+		/// <param name="progress"></param>
+		/// <param name="bookServer"></param>
+		/// <param name="startingMessageAction"></param>
+		public static void SendBook(Book.Book book, BookServer bookServer, string destFileName, Action<string, string> sendAction, WebSocketProgress progress, Func<string, string, string> startingMessageAction,
```

Comment:
If there's reason to open this up again, startingMessageAction --> startingMessageFunction or getStartingMesssage